rootfs/scheduler/__init__.py

```diff
@@ -1004,12 +1004,7 @@ def _scale_rc(self, namespace, name, desired):
             'version': rc['spec']['selector']['version']
         }
 
-        # Are there any pods running (and verified as ready) available?
-        pods = self._get_pods(namespace, labels=labels).json()['items']
-        current = 0
-        for pod in pods:
-            if self._pod_ready(pod):
-                current += 1
+        current = int(rc['spec']['replicas'])
```

Comment:
The `desired` value is already passed in, which is the value that is saved as `rc['spec']['replicas']`

See https://github.com/deis/controller/blob/master/rootfs/scheduler/__init__.py#L1019 and then https://github.com/deis/controller/blob/master/rootfs/scheduler/__init__.py#L1050

Your code is then doing `current = rc['spec']['replicas']` which is just the same value as `desired` will already be using